youtube_dl/extractor/vtm.py

```diff
@@ -0,0 +1,133 @@
+from ..utils import (
+    urlencode_postdata,
+    compat_urllib_parse_urlencode,
+    ExtractorError,
+)
+
+
+class VTMIE(InfoExtractor):
+    """Download full episodes that require an account from vtm.be or q2.be.
+
+    The generic extractor can be used to download clips that do no require an
+    account.
+    """
+    _VALID_URL = r'https?://(?:www\.)?(?P<site_id>vtm|q2)\.be/video[/?].+?'
+    _NETRC_MACHINE = 'vtm'
+    _APIKEY = '3_HZ0FtkMW_gOyKlqQzW5_0FHRC7Nd5XpXJZcDdXY4pk5eES2ZWmejRW5egwVm4ug-'
+    _TESTS = [
+        {
+            'url': 'http://vtm.be/video/volledige-afleveringen/id/vtm_20170219_VM0678361_vtmwatch',
+            'md5': '9357926318cdfcb10079d85d56f4e04e',
+            'info_dict': {
+                'id': 'vtm_20170219_VM0678361_vtmwatch',
+                'ext': 'mp4',
+                'title': 'Allemaal Chris afl. 6',
+                'description': 'md5:4be86427521e7b07e0adb0c9c554ddb2',
+            },
+            'skip_download': True,
+        }
+    ]
+
+    def _login(self):
+        (username, password) = self._get_login_info()
+        if username is None or password is None:
+            self.raise_login_required()
+
+        auth_data = {
+            'APIKey': self._APIKEY,
+            'sdk': 'js_6.1',
+            'format': 'json',
+            'loginID': username,
+            'password': password,
+        }
+
+        auth_info = self._download_json(
+            'https://accounts.eu1.gigya.com/accounts.login', None,
+            note='Logging in', errnote='Unable to log in',
+            data=urlencode_postdata(auth_data), fatal=True)
+
+        error_message = auth_info.get('errorDetails')
+        if error_message:
+            raise ExtractorError(
+                'Unable to login: %s' % error_message, expected=True)
+
+        self._uid = auth_info.get('UID')
+        self._uid_signature = auth_info.get('UIDSignature')
+        self._signature_timestamp = auth_info.get('signatureTimestamp')
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        site_id = mobj.group('site_id')
+
+        webpage = self._download_webpage(url, None)
+
+        # There are two known types of URLs both pointing to the same video.
+        # The type of URL you get depends on how you navigate to the video.
+        # Unfortunately, only one type of URL contains the actual video id:
+        #
+        # http://vtm.be/video/volledige-afleveringen/id/257059160170000
+        #
+        # The video id can also be something like:
+        # "vtm_20170213_VM0678492_vtmwatch"
+        #
+        # The other type of URL looks like:
+        #
+        # http://vtm.be/video?aid=166380
+        #
+        # The 'aid' is not the video id we are looking for. There can also be
+        # a lot of very long parameters in addition to the 'aid' parameter.
+        #
+        # So to work in all cases, we extract the video_id from the webpage
+        # instead of the URL.
```

Comment:
Ok. Maybe you can help me with the following: when I add multiple tests, only the first one seems to be run. I use `python test/test_download.py TestDownload.test_VTM` to run the tests. When I try running the tests of another extractor, e.g. `Youtube`, the same happens.